src/Framework/Framework/Controls/HtmlGenericControl.cs

```diff
@@ -412,19 +413,25 @@ private void AddHtmlAttributesToRender(ref RenderState r, IHtmlWriter writer)
 
                 if (knockoutExpression is {})
                 {
-                    if (gprop.GroupMemberName == "class")
+                    if (attributeName == "class")
                     {
                         writer.AddKnockoutDataBind("class", knockoutExpression);
                     }
                     else
                     {
                         attributeBindingGroup ??= new KnockoutBindingGroup();
-                        attributeBindingGroup.Add(gprop.GroupMemberName, knockoutExpression);
+                        attributeBindingGroup.Add(attributeName, knockoutExpression);
                     }
                     if (!r.RenderOnServer(this))
                         continue;
                 }
-                AddHtmlAttribute(writer, gprop.GroupMemberName, valueRaw);
+                AddHtmlAttribute(writer, attributeName, valueRaw);
+
+                if (attributeName == "id")
```

Comment:
We should ensure that `attributeName` is always lower-case and invariant culture :)